-1

Tested at: 5ce73435097b8f423dd5feba33bb6bedd60d522f

CMSSW: CMSSW_11_2_X_2020-09-12-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c2f0e6/9287/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found compilation error when building: 

<pre>Copying tmp/slc7_amd64_gcc820/src/HeavyFlavorAnalysis/RecoDecay/src/HeavyFlavorAnalysisRecoDecay/libHeavyFlavorAnalysisRecoDecay.so to productstore area:
Leaving library rule at HeavyFlavorAnalysis/RecoDecay
Entering library rule at src/HeavyFlavorAnalysis/RecoDecay/test
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-09-12-1100/src/HeavyFlavorAnalysis/RecoDecay/test/stubs/TestBPHRecoDecay.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-09-12-1100/src/HeavyFlavorAnalysis/RecoDecay/test/stubs/TestBPHRecoDecay.cc: In member function 'virtual bool TestBPHRecoDecay::analyze(const edm::Event&, const edm::EventSetup&)::Chi2Select::accept(const BPHDecayVertex&) const':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-09-12-1100/src/HeavyFlavorAnalysis/RecoDecay/test/stubs/TestBPHRecoDecay.cc:313:22: error: 'Prob' is not a member of 'TMath'
       return (TMath::Prob(v.chi2(), lround(v.ndof())) > mProb);
                      ^~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-09-12-1100/src/HeavyFlavorAnalysis/RecoDecay/test/stubs/TestBPHRecoDecay.cc: In member function 'void TestBPHRecoDecay::dumpRecoCand(const string&, const BPHRecoCandidate*)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-09-12-1100/src/HeavyFlavorAnalysis/RecoDecay/test/stubs/TestBPHRecoDecay.cc:510:24: error: 'Prob' is not a member of 'TMath'
   double prob = TMath::Prob(chi2, ndof);
</pre>

--- a/DataFormats/MuonReco/src/MuonSelectors.cc
+++ b/DataFormats/MuonReco/src/MuonSelectors.cc
@@ -220,9 +220,9 @@ float muon::segmentCompatibility(const reco::Muon& muon, reco::Muon::Arbitration
                 // only use pull if 3 sigma is not smaller than 3 cm
                 if (muon.dX(i, 1, arbitrationType) < 3.f && muon.pullX(i, 1, arbitrationType) > 3.f) {
                   if (muon.dX(i, 1, arbitrationType) > 1.f)
-                    station_weight[i - 1] *= 1.f / std::pow(muon.dX(i, 1, arbitrationType), .125);
+                    station_weight[i - 1] *= 1.f / std::pow(muon.dX(i, 1, arbitrationType), .25);
                 } else {
-                  station_weight[i - 1] *= 1.f / std::pow(muon.pullX(i, 1, arbitrationType), .125);
+                  station_weight[i - 1] *= 1.f / std::pow(muon.pullX(i, 1, arbitrationType), .25);
                 }
               }
             }
@@ -234,9 +234,9 @@ float muon::segmentCompatibility(const reco::Muon& muon, reco::Muon::Arbitration
                 // only use pull if 3 sigma is not smaller than 3 cm
                 if (muon.dY(i, 1, arbitrationType) < 3. && muon.pullY(i, 1, arbitrationType) > 3.) {
                   if (muon.dY(i, 1, arbitrationType) > 1.f)
-                    station_weight[i - 1] *= 1.f / std::pow(muon.dY(i, 1, arbitrationType), .125);
+                    station_weight[i - 1] *= 1.f / std::pow(muon.dY(i, 1, arbitrationType), .25);
                 } else {
-                  station_weight[i - 1] *= 1.f / std::pow(muon.pullY(i, 1, arbitrationType), .125);
+                  station_weight[i - 1] *= 1.f / std::pow(muon.pullY(i, 1, arbitrationType), .25);
                 }
               }
             }
@@ -250,7 +250,7 @@ float muon::segmentCompatibility(const reco::Muon& muon, reco::Muon::Arbitration
               const float dxy2 =
                   std::pow(muon.dX(i, 1, arbitrationType), 2.) + std::pow(muon.dY(i, 1, arbitrationType), 2.);
               // only use pull if 3 sigma is not smaller than 3 cm
-              if (dxy2 < 9.f && pullTot2 < 9.f) {
+              if (dxy2 < 9.f && pullTot2 > 9.f) {
                 if (dxy2 > 1.f)
                   station_weight[i - 1] *= 1.f / std::pow(dxy2, .125);
               } else {

--- a/HeavyFlavorAnalysis/RecoDecay/test/stubs/TestBPHRecoDecay.cc
+++ b/HeavyFlavorAnalysis/RecoDecay/test/stubs/TestBPHRecoDecay.cc
@@ -21,6 +21,7 @@
 #include "RecoVertex/KinematicFit/interface/TwoTrackMassKinematicConstraint.h"
 #include <TH1.h>
 #include <TFile.h>
+#include <TMath.h>
 
 #include <set>
 #include <string>

--- a/PhysicsTools/PatAlgos/src/SoftMuonMvaEstimator.cc
+++ b/PhysicsTools/PatAlgos/src/SoftMuonMvaEstimator.cc
@@ -7,6 +7,8 @@
 #include "DataFormats/MuonReco/interface/MuonSelectors.h"
 #include "DataFormats/PatCandidates/interface/Muon.h"
 
+#include "TMath.h"
+
 using namespace pat;
 
 SoftMuonMvaEstimator::SoftMuonMvaEstimator(const edm::FileInPath& weightsfile) {

